src/main/java/org/springframework/social/openidconnect/HttpClientFactory.java

```diff
@@ -114,4 +116,26 @@ private static X509HostnameVerifier getVerifier(boolean isStrict){
         return SSLSocketFactory.STRICT_HOSTNAME_VERIFIER;
     }
 
+    /**
+     * Gets the TLS version from jdk parameter.  Defaults to version 1.2 for JDK 1.7 if not provided
+     * @return
+     */
+    public static String getTlsVersion() {
+        // don't use "jdk.tls.client.protocols" as it was causing unexpected exception during unit testing
+        String tlsVer = System.getProperty("tlsProtocol");
+
+        if(StringUtils.isEmpty(tlsVer)) {
+            double javaVersion = Double.parseDouble(System.getProperty("java.specification.version"));
+            if(javaVersion == 1.7) {
```

Comment:
@abhijith-prabhakar after discussion, we'll hardcode the default to TLSv1.2 as you suggested, since this module will only be used to connect to PayPal endpoints that are ending TLSv1.0 support (breaking JDK <= 1.6), and coding against JDK version is considered bad practice.